internal/bridge/managedpostgrescluster/managedpostgrescluster_controller.go

```diff
@@ -0,0 +1,524 @@
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+
+			if !deletedAlready {
+				return ctrl.Result{RequeueAfter: 1 * time.Second}, err
+			}
+
+			// Remove finalizer if deleted already
+			if deletedAlready {
+				log.Info("cluster deleted", "clusterName", managedpostgrescluster.Spec.ClusterName)
+
+				controllerutil.RemoveFinalizer(managedpostgrescluster, finalizer)
+				if err := r.Update(ctx, managedpostgrescluster); err != nil {
+					return ctrl.Result{}, err
+				}
+			}
+		}
+		// Stop reconciliation as the item is being deleted
+		return ctrl.Result{}, nil
+	}
+
+	// Wonder if there's a better way to handle adding/checking/removing statuses
+	// We did something in the upgrade controller
+	// Exit early if we can't create from this K8s object
+	// unless this K8s object has been changed (compare ObservedGeneration)
+	invalid = meta.FindStatusCondition(managedpostgrescluster.Status.Conditions,
+		v1beta1.ConditionCreating)
+	if invalid != nil &&
+		invalid.Status == metav1.ConditionFalse &&
+		invalid.Message == "ClusterInvalid" &&
+		invalid.ObservedGeneration == managedpostgrescluster.GetGeneration() {
+		return ctrl.Result{}, nil
+	}
+
+	// Remove cluster invalid status if found
+	if invalid != nil &&
+		invalid.Status == metav1.ConditionFalse &&
+		invalid.Reason == "ClusterInvalid" {
+		meta.RemoveStatusCondition(&managedpostgrescluster.Status.Conditions,
+			v1beta1.ConditionCreating)
+	}
+
+	storageVal, err := handleStorage(managedpostgrescluster.Spec.Storage)
+	if err != nil {
+		log.Error(err, "whoops, storage issue")
+		// TODO(managedpostgrescluster)
+		// lint:ignore nilerr no requeue needed
+		return ctrl.Result{}, nil
+	}
+
+	// We should only be missing the ID if no create has been issued
+	// or the create was interrupted and we haven't received the ID.
+	if managedpostgrescluster.Status.ID == "" {
+		// START FIND
+
+		// TODO(managedpostgrescluster) If the CreateCluster response was interrupted, we won't have the ID
+		// so we can get by name
+		// BUT if we do that, there's a chance for the K8s object to grab a pre-existing Bridge cluster
+		// which means there's a chance to delete a Bridge cluster through K8s actions
+		// even though that cluster didn't originate from K8s.
+
+		// Check if the cluster exists
+		clusters, err := r.NewClient().ListClusters(ctx, key, team)
+		if err != nil {
+			log.Error(err, "whoops, cluster listing issue")
+			return ctrl.Result{}, err
+		}
+
+		for _, cluster := range clusters {
+			if managedpostgrescluster.Name == cluster.Name {
+				managedpostgrescluster.Status.ID = cluster.ID
+				// Requeue now that we have a cluster ID assigned
+				return ctrl.Result{Requeue: true}, nil
+			}
+		}
+
+		// END FIND
+
+		// if we've gotten here then no cluster exists with that name and we're missing the ID, ergo, create cluster
+
+		// TODO(managedpostgrescluster) Can almost just use the managed.Spec... except for the team, which we don't want
+		// users to set on the spec. Do we?
```

Comment:
I was wondering about this myself... Not sure the team ID really needs to be kept secret. And now that we know that API Keys can have access to multiple teams, if we move the Team ID to the spec, then users don't need a separate Secret for each team.

Reply:
Ticket made for that, so resolving this